server/src/routes/user.ts

```diff
@@ -0,0 +1,68 @@
+import { PrismaClient, UserRole } from "@prisma/client";
+import express from "express";
+
+export const userRoutes = express.Router();
+const prisma = new PrismaClient();
+
+// Filter using query string in url with parameters role and category
+userRoutes.route("/").get(async (req, res) => {
+  let categoryFilter = (req.query.category as string) || null;
+  let roleFilter = (req.query.role as string) || null;
+  let filter: any = {};
+
+  if (categoryFilter !== null) {
+    const categoryId = parseInt(categoryFilter);
+    if (isNaN(categoryId)) {
+      res.status(400).send("Invalid category query parameter, must be an integer");
+      return;
+    } else {
+      filter.categoryGroupId = categoryId;
+    }
+  }
+
+  if (roleFilter !== null) {
+    roleFilter = roleFilter.toUpperCase();
+    if (!(roleFilter in UserRole)) {
+      res.status(400).send("Invalid user role");
+      return;
+    } else {
+      filter.role = roleFilter;
+    }
+  }
```

Comment:
The error handling on this is really great! Since this is a backend server that's just communicating with our frontend server, we don't really need that much error handling though, since we expect that most of our requests will be formatted correctly. As such, you can mostly remove all this error handling, and let prisma throw an error if something is off.